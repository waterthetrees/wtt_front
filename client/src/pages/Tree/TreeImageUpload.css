.not-dragging {
  background-color: none;
}

.dragging {
  background-color: gray;
}