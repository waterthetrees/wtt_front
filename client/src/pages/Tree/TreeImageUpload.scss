$color: black;
$font: "Montserrat";

// Per Google Bard:
// If you want to use a custom class name
// that is not already defined by Material UI,
// you will need to double the class name.
//
// This is because Material UI uses a prefix for its class names.
// The prefix is usually Mui followed by the name of the component.
// For example, the class name for the Button component is MuiButton.
// When you double the class name, you are telling Material UI that
// you want to use a custom class, not a class that is defined by Material UI.

.treeimageupload.treeimageupload {
  margin: 1em 0;
}

.text.text {
  color: $color;
  font-size: 12px;
  font-family: $font;
  margin: 1px;
}

.image-upload-area.image-upload-area {
  margin: .5em;
  display: flex;
  justify-content: center;
}

.drop-area.drop-area {
  display: flex;
  flex-direction: column;
  place-items: center;
  justify-content: center;
  width: 100%;
  border-radius: 8px;
  border: 2px dashed $color;
  text-align: center;
  font-family: $font;
}

.action-text.action-text {
  color: $color;
  font-family: $font;
  font-size: 16px;
  font-style: medium;
  margin: 5px;
}

.photo-icon.photo-icon {
  color: #7d7d7d;
  width: 32px;
  height: 32px;
  margin: 5px 0 1px;
}

.input-border.input-border {
  display: flex;
  font-family: Montserrat;
  align-items: center;
  background-color: #ededed;
  border-radius: 8px;
  justify-content: center;
  padding: 4px 2px;
}

.dialog-button.dialog-button {
  border-radius: 8px;
  font-size: 12px;
  background-color: #fff;
  color: $color;
  padding: 0.1em 1em;
  text-transform: none;

  &:hover {
    background-color: #ececec;
  }
}

.error-text.error-text {
  color: red;
  font-size: 12px;
  font-family: $font;
  margin: 1px;
}

.upload-bar.upload-bar {
  margin: 1em 0;
  display: grid;
  grid-template-columns: 1fr 5fr 2fr;
  align-items: center;
  grid-gap: 1em;
}

.not-dragging.not-dragging {
  background-color: none;
}

.dragging.dragging {
  background-color: rgb(0 0 0 / 4%);
}
