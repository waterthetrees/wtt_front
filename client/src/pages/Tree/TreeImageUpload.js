import React, { useRef } from 'react';
import { AddAPhoto } from "@mui/icons-material";
import { Button, IconButton, LinearProgress, styled, TextField } from "@mui/material";
import { useEffect, useState } from "react";
import "./TreeImageUpload.css";

const color = "#9d9d9d";
const font = "Montserrat";

const ImageUploadSection = styled('section')`
  margin: 1em 0;
`;

const Border = styled(IconButton)`
  display: flex;
  flex-direction: column;
  place-items: center;
  justify-content: center;
  margin: auto;
  padding: 1em;
  width: 100%;
  border-radius: 8px;
  border: 2px dashed ${color};
  text-align: center;
  font-family: Montserrat;
`;

const ActionText = styled('h2')`
  color: ${color};
  font-family: ${font};
  font-size: 16px;
  font-style: medium;
  margin: 5px;
`;

const Icon = styled(AddAPhoto)`
  color: #7d7d7d;
  width: 32px;
  height: 32px;
  margin: 5px 0 1px;
`;

const Text = styled('p')`
  color: ${color};
  font-size: 12px;
  font-family: ${font};
  margin: 1px;
`;

const HiddenInput = styled('input')`
  display: none;
`;

function ImageUploadArea({ uploadURL, fileDialog }) {
  const [dragActive, setDragActive] = useState(false);
  const handleDrag = (e) => {
    e.preventDefault();
    e.stopPropagation();
    if (e.type === "dragenter" || e.type === "dragover") {
      setDragActive(true);
    } else if (e.type === "dragleave") {
      setDragActive(false);
    }
  };

  const inputRef = useRef(null);

  const handleClick = () => {
    inputRef.current.click();
  };

  const fileSelected = () => {
    uploadURL(inputRef.current.files[0].name);
  };

  const addImage = (e) => {
    e.preventDefault();
    e.stopPropagation();
    uploadURL(e.dataTransfer.files[0].name);
    setDragActive(false);
  }

  return (
    <>
      <Border
        onDragEnter={handleDrag}
        onDragOver={handleDrag}
        onDragLeave={handleDrag}
        onDrop={addImage}
        onClick={handleClick}
        className={dragActive ? "dragging" : "not-dragging"}
      >
          <Icon />
          <ActionText>
            Select Image to Upload
          </ActionText>
          <Text>or drag and drop here</Text>
      </Border>
      <HiddenInput type="file" ref={inputRef} onChange={fileSelected} />
    </>
  );
}

const InputBorder = styled('section')`
  display: flex;
  font-family: Montserrat;
  align-items: center;
  background-color: #ededed;
  border-radius: 8px;
  justify-content: center;
  padding: 4px 2px;
`;

const DialogButton = styled(Button)`
  border-radius: 8px;
  font-size: 12px;
  background-color: #ffffff;
  color: black;
  padding: .1em 1em;
  text-transform: none;

  &:hover {
    background-color: #ececec;
  }
`;

const ErrorText = styled('p')`
  color: red;
  font-size: 12px;
  font-family: ${font};
  margin: 1px;
`;

function ImageUploadDialog({ uploadURL }) {
  const [value, setValue] = useState("");
  const [error, setError] = useState(false);

  const handleClick = () => {
    if (!value) {
      setError(true);
      return;
    }

    uploadURL(value);
    setValue("");
    setError(false);
  }

  const handleChange = (e) => {
    setValue(e.target.value);
  }

  return (
    <>
      <InputBorder>
        <TextField
          hiddenLabel
          InputProps={{ disableUnderline: true }}
          placeholder='Add a file URL'
          variant='standard'
          value={value}
          onChange={handleChange}
        />
        <DialogButton
          variant='contained'
          onClick={handleClick}
        >
          Upload
        </DialogButton>
      </InputBorder>
      {error && <ErrorText>Invalid URL</ErrorText>}
    </>
  );
}

const UploadBar = styled("div")`
  margin: 1em 0;
  display: grid;
  grid-template-columns: 1fr 5fr 2fr;
  align-items: center;
  grid-gap: 1em;
`;

const Percent = styled("p")`
  margin: auto;
`;

function ImageUploadBar({ pollUpload, uploadDone, cancelUpload }) {
  const [progress, setProgress] = useState(0);

  useEffect(() => { 
    pollUpload((newProgess) => setProgress(newProgess), uploadDone);
  }, []);

  return (
    <UploadBar>
      <Percent>{progress}%</Percent>
      <LinearProgress
        variant="determinate"
        color="success"
        value={progress}
      />
      <DialogButton
        variant="contained"
        onClick={cancelUpload}
      >
        Cancel
      </DialogButton>
    </UploadBar>
  );
}

export default function ImageUpload({
  apiUploadURL,
  apiUploadDone,
  apiPollUpload,
  apiCancelUpload,
  apiFileDialog,
  isMobile
}) {
  const [isUploading, setIsUploading] = useState(false);

  function uploadURL(url) {
    apiUploadURL(url)
    setIsUploading(true);
  }

  function uploadDone() {
    apiUploadDone();
    setIsUploading(false);
  }

  function cancelUpload() {
    apiCancelUpload();
    setIsUploading(false);
  }

  return (
    <ImageUploadSection>
      <h3>Uploading Images</h3>
      <p sx={{ fontSize: "12px" }}>
        Took a selfie with your tree and want
        to track the life of your tree from
        the day you started.
      </p>
      <ImageUploadArea
        uploadURL={uploadURL}
        fileDialog={apiFileDialog}
      />
      {!isMobile &&
        <>
          <p>Or upload from a URL</p>
          <ImageUploadDialog uploadURL={uploadURL} />
        </>
      }
      {isUploading &&
        <ImageUploadBar
          uploadDone={uploadDone}
          cancelUpload={cancelUpload}
          pollUpload={apiPollUpload}
        />
      }
    </ImageUploadSection>
  )
}