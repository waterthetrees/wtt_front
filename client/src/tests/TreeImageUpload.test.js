import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TreeImageUpload from '../pages/Tree/TreeImageUpload';

const mockApiUploadURL = jest.fn();
const mockApiUploadDone = jest.fn();
const mockApiCancelUpload = jest.fn();
// A mock API for polling upload.
const mockApiPollUpload = jest.fn((progressHook, uploadDone) => {
  let progress = 0;
  let intervalId = setInterval(
    () => {
      progress = progress + 10; 
      progressHook(progress);
      if (progress >= 100) {
        clearInterval(intervalId);
        intervalId = null;
        uploadDone();
      }
    },
    100
  );
});

const ImageUpload = ({ isMobile }) =>
  <TreeImageUpload
    apiUploadURL={mockApiUploadURL}
    apiUploadDone={mockApiUploadDone}
    apiCancelUpload={mockApiCancelUpload}
    apiPollUpload={mockApiPollUpload}
    isMobile={isMobile}
  />;

describe("TreeImageUpload", () => {
  it("renders correctly on desktop", () => {
    const { container } = render(<ImageUpload />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly on mobile", () => {
    const { container } = render(<ImageUpload isMobile={true} />);
    expect(container).toMatchSnapshot();
  });

  it("can upload from dialog", async () => {
    const user = userEvent.setup();
    const { getByRole, queryByRole } = render(<ImageUpload />);
    const dialogTextField = getByRole("textbox");
    const dialogButton = getByRole("button", { name: "Upload" });

    // Click on the upload button.
    await user.click(dialogTextField);
    await user.keyboard("fakefile.jpg");
    await user.click(dialogButton);
    expect(mockApiUploadURL).toHaveBeenCalled();

    await waitFor(() => {
      // I don't test for the upload bar,
      // but I assume if the cancel button
      // is there then so is it.
      const cancelButton = getByRole("button", { name: "Cancel" });
      expect(cancelButton).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(mockApiUploadDone).toHaveBeenCalled();
      const cancelButton = queryByRole("button", { name: "Cancel" });
      expect(cancelButton).not.toBeInTheDocument();
    });

    // Or press enter.
    await user.click(dialogTextField);
    await user.keyboard("fakefile.jpg{Enter}");
    expect(mockApiUploadURL).toHaveBeenCalled();
  });

  it("can cancel upload", async () => {
    const user = userEvent.setup();
    const { getByRole, queryByRole } = render(<ImageUpload />);
    const dialogTextField = getByRole("textbox");
    const dialogButton = getByRole("button", { name: "Upload" });
    await user.click(dialogTextField);
    await user.keyboard("fakefile.jpg");
    await user.click(dialogButton);
    expect(mockApiUploadURL).toHaveBeenCalled();
    await waitFor(() => {
      const cancelButton = getByRole("button", { name: "Cancel" });
      expect(cancelButton).toBeInTheDocument();
      user.click(cancelButton);
    });
    await waitFor(() => {
      expect(mockApiCancelUpload).toHaveBeenCalled();
      const cancelButton = queryByRole("button", { name: "Cancel" });
      expect(cancelButton).not.toBeInTheDocument();
    });
  });

  it("will not upload if not a valid URL", async () => {
    const user = userEvent.setup();
    const { getByRole, getByText } = render(<ImageUpload />);
    const dialogButton = getByRole("button", { name: "Upload" });
    await user.click(dialogButton);
    await waitFor(() =>
      expect(getByText(/Only jpg/)).toBeInTheDocument()
    );
    await user.keyboard("fakefile.png");
    await user.click(dialogButton);
    await waitFor(() =>
      expect(getByText(/Only jpg/)).toBeInTheDocument()
    );
  });

  it("can upload from drag and drop", async () => {
    const { getByRole } = render(<ImageUpload />);
    const dropzone = getByRole("button", { name: /Select Image to Upload/ });
    fireEvent.drop(dropzone, { dataTransfer: { files: [ { name: "badfile.png" }] } });
    expect(mockApiUploadURL).not.toHaveBeenCalled();
    fireEvent.drop(dropzone, { dataTransfer: { files: [ { name: "goodfile.jpg" }] } });
    expect(mockApiUploadURL).toHaveBeenCalled();
  });

  it("can upload by bringing up file dialog", async () => {
    const user = userEvent.setup();
    const badFile = new File([""], "fakefile.png", { type: "image/png" });
    const goodFile = new File([""], "fakefile.jpg", { type: "image/jpeg" });
    const { getByTestId } = render(<ImageUpload />);
    const input = getByTestId("file-input");
    
    await user.upload(input, badFile);
    expect(input.files).toHaveLength(0);

    await user.upload(input, goodFile);
    expect(input.files).toHaveLength(1);
  });
});