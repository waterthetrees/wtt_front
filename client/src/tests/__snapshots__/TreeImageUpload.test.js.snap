// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeImageUpload renders correctly on desktop 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-4 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-4.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

label+.emotion-4 {
  margin-top: 16px;
}

.emotion-5 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-5::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-5:focus {
  outline: 0;
}

.emotion-5:invalid {
  box-shadow: none;
}

.emotion-5::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-5:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-5.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-5:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: #1976d2;
  }
}

.emotion-6:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-6.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-6.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

<div>
  <section
    sx="[object Object]"
  >
    <h3>
      Uploading Images
    </h3>
    <p
      sx="[object Object]"
    >
      Took a selfie with your tree and want to track the life of your tree from the day you started.
    </p>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium drop-area not-dragging emotion-0"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium photo-icon emotion-1"
        data-testid="AddAPhotoIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
        />
      </svg>
      <h2
        class="action-text"
      >
        Select Image to Upload
      </h2>
      <p
        class="text"
      >
        or drag and drop here
      </p>
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <input
      accept="image/jpeg"
      data-testid="file-input"
      sx="[object Object]"
      type="file"
    />
    <p>
      Or upload from a URL
    </p>
    <section
      class="input-border"
    >
      <div
        class="MuiFormControl-root MuiTextField-root emotion-3"
      >
        <div
          class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-hiddenLabel emotion-4"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input MuiInputBase-inputHiddenLabel emotion-5"
            id=":r0:"
            placeholder="Add a file URL"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium dialog-button emotion-6"
        tabindex="0"
        type="button"
      >
        Upload
        <span
          class="MuiTouchRipple-root emotion-2"
        />
      </button>
    </section>
  </section>
</div>
`;

exports[`TreeImageUpload renders correctly on mobile 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <section
    sx="[object Object]"
  >
    <h3>
      Uploading Images
    </h3>
    <p
      sx="[object Object]"
    >
      Took a selfie with your tree and want to track the life of your tree from the day you started.
    </p>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium drop-area not-dragging emotion-0"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium photo-icon emotion-1"
        data-testid="AddAPhotoIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
        />
      </svg>
      <h2
        class="action-text"
      >
        Select Image to Upload
      </h2>
      <p
        class="text"
      >
        or drag and drop here
      </p>
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <input
      accept="image/jpeg"
      data-testid="file-input"
      sx="[object Object]"
      type="file"
    />
  </section>
</div>
`;
